examples/cached_pool.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the MIT License, <LICENSE or http://opensource.org/licenses/MIT>.
+// This file may not be copied, modified, or distributed except according to those terms.
+
+extern crate env_logger;
+#[macro_use]
+extern crate log;
+extern crate tarpc;
+
+use std::thread;
+use std::time::Duration;
+use tarpc::cached_pool::{CachedPool, Config};
+
+fn main() {
+    let _ = env_logger::init();
+
+    let pools = &[CachedPool::new(Config::max_idle(Duration::from_secs(5))),
+                  CachedPool::new(Config::max_idle(Duration::from_millis(500)))];
+    for _ in 0..15 {
+        for pool in pools {
+            pool.execute(move || {
+                    thread::sleep(Duration::from_secs(5));
+                })
+                .unwrap();
+        }
+        info!("{:?}",
```

Comment:
Gonna push back a bit on this, because I don't think we should comment all iterator chained methods. Can you elaborate on what's confusing you so I have a better idea of what needs commenting?